doc/examples/cloud-config-puppet.txt

```diff
@@ -1,93 +0,0 @@
-#cloud-config
```

Comment:
Why remove the examples?

Reply:
I mentioned it in the PR description:
```I also removed files from doc/examples that were a single example. It serves as a secondary, less visible spot for docs and examples, when the examples could just as easily go into the documentation.```

I've noticed a lot of modules have an example, then there's an example file with almost the exact same example with comments that more of less serve as documentation. Seems it'd be best to have that all in one place, and save example files for times when we have multiple different examples for multiple different use cases.